src/Language/PureScript/Errors.hs

```diff
@@ -1318,7 +1319,7 @@ toTypelevelString t = (Box.text . decodeStringWithReplacement) <$> toTypelevelSt
   toTypelevelString' :: Type -> Maybe PSString
   toTypelevelString' (TypeLevelString s) = Just s
   toTypelevelString' (TypeApp (TypeConstructor f) x)
-    | f == primName "TypeString" = Just $ fromString $ prettyPrintType x
+    | f == primName "TypeString" = fromString <$> (initMay . Box.render $ typeAsBox x)
```

Comment:
`prettyPrintType` was already calling `Box.render`. But yes, it's a hack. I'll think about it more.